### ISSUE TITLE
Support multiple parametrized statements in single SQL Object extension

### ISSUE THREAD

**Issue by findepi** [jdbi/jdbi#1307]
Intended use, something like
```
@SqlScript("DELETE FROM relation_table WHERE my_id = :id; " +
   "DELETE FROM my_table WHERE my_id = :id")
void delete(String id);
```

Currently (jdbi 3.5.1), `SqlScript` does not replace `:<params>` and passes them literal to JDBC.

**Comment by findepi** [comment]
Note: `@SqlScript` does allow `<name>`-style parametrization with `@Define` annotation. Clearly, this is not what i would need. I need something that would map to prepared statement's `?`.

**Comment by TheRealMarnes** [comment]
Are there 2 distinct issues here? 1) allow multiple statements, and 2) support :params in SqlScript (and implicitly, let :params work in scripts with multiple statements)? It kinda sounds like 1 is already OK, that your issue is only that :params don't work in SqlScript.

I haven't done any observations in our code or tests yet, just clarifying the issue before any discussion.

**Comment by stevenschlansker** [comment]
I myself ran into a use case for a single bind+define annotation as well.  That seems like a reasonable feature request.

Regarding multiple update annotations, what would you expect the final result to be?  We will have a potentially arbitrary number of result sets or update counts to map.  We could attempt to provide access to all result sets, but this could be a little tricky...

I'm wary of splitting on `;`.  Right now we map nicely 1 to 1, one `@SqlUpdate` is exactly one `PreparedStatement` -- changing this likely has huge consequences.

**Comment by findepi** [comment]
> I'm wary of splitting on `;`. ...

I agree. I also would prefer to avoid `;`-based splitting when possible.

Altenative syntaxes:
```
@SqlUpdate({
   "first query",
   "second query"})
void runTwoUpdates();
```

```
@SqlUpdate("first query")
@SqlUpdate("second query")
void runTwoUpdates();
```

As having multiple annotations is possible but not widely established (at least from my point of view), the first option, with `@SqlUpdate.value` being an array, might be easier discoverable by users.

> Regarding multiple update annotations, what would you expect the final result to be?

Good question. Assuming single update produces single update count, either `void` (return none) or `List<Long>`  (return all update counts) or `long` (return last) would do.
(Obviously, single sql stmt can produce more than 1 update count, but that's is problem already, isn't it?)

For the record, i was able to workaround / implement this differently leveraging http://jdbi.org/#_default_methods and i'm pretty OK with this, as long as I don't have to do this too often (which i don't):

```java
interface MySqlObject extends /*org.jdbi.v3.sqlobject.*/ SqlObject {

    default void delete(String id) {
        useHandle(handle -> {
            handle.createUpdate("DELETE FROM relation_table WHERE my_id = :id; ").bind("id", id).execute();
            handle.createUpdate("DELETE FROM my_table WHERE my_id = :id").bind("id", id).execute();
        });
    }
}
```

Having this explicit avoids the problem of return value -- this can be whatever i want.

**Comment by qualidafial** [comment]
The two features I'm seeing proposed are:

- Make `@SqlUpdate` repeatable
- Make `@SqlScript` support bound parameters.

I think either of these would be fine, but doing both would deliver essentially the same functionality twice in the API.

I lean toward making `@SqlUpdate` repeatable. Potentially we could deprecate `@SqlScript` at the same time?

**Comment by stevenschlansker** [comment]
Unfortunately I don't think the semantics here make sense. There's no clean return contract, since methods only return a single value, and the parameter binding does not naturally work with Script either. Using default methods to call multiple statements is the supported way to do this for now. I landed an update to clarify this in the docs.
